src/cloud/social/provider.ts

```diff
@@ -391,9 +391,19 @@ export class CloudSocialProvider {
         new Error('blockUser unimplemented'));
   }
 
-  public removeUser = (userId: string): Promise<void> => {
-    return Promise.reject(
-        new Error('removeUser unimplemented'));
+  // Removes a cloud contact from storage
+  public removeUser = (host: string): Promise<void> => {
+    log.debug('removing cloud contact %1 from storage %2', host, STORAGE_KEY);
+    if (!this.savedContacts_[host]) {
+      // Do not return an error because result is as expected.
+      log.error('cloud contact %1 is not in %2 - cannot remove from storage', host, STORAGE_KEY);
+    } else {
+      // Remove host from this.savedContacts_
+      delete this.savedContacts_[host];
+      // Update storage with this.savedContacts_
+      this.saveContacts_();
+    }
+    return Promise.resolve<void>();
```

Comment:
I was going to suggest you just `return this.saveContacts_()` here and realised I implemented a classic "swallow and log" anti-pattern...my bad. Do you mind changing `saveContacts_` to read:
```typescript
  // Saves contacts to storage.
  private saveContacts_ = () => {
    log.debug('saveContacts');
    return this.storage_.set(STORAGE_KEY, JSON.stringify(<SavedContacts>{
      contacts: Object.keys(this.savedContacts_).map(key => this.savedContacts_[key])
    });
  }
```
And **then** have this new function `return this.saveContacts_`?